pandas/core/arrays/arrow/_arrow_utils.py

```diff
@@ -14,10 +16,11 @@ def fallback_performancewarning(version: str | None = None) -> None:
     Raise a PerformanceWarning for falling back to ExtensionArray's
     non-pyarrow method
     """
-    msg = "Falling back on a non-pyarrow code path which may decrease performance."
-    if version is not None:
-        msg += f" Upgrade to pyarrow >={version} to possibly suppress this warning."
-    warnings.warn(msg, PerformanceWarning, stacklevel=find_stack_level())
+    if _get_option("performance_warnings"):
+        msg = "Falling back on a non-pyarrow code path which may decrease performance."
+        if version is not None:
+            msg += f" Upgrade to pyarrow >={version} to possibly suppress this warning."
+        warnings.warn(msg, PerformanceWarning, stacklevel=find_stack_level())
```

Comment:
Going forward how can we ensure that `PerformanceWarning`s are issued without checking the option?

Reply:
We currently raise if a warning is emitted in a test outside of `tm.assert_produces_warning`.

Reply:
True, as long as we have a test for it :)